tests/test_virtualenv.py

```diff
@@ -279,6 +279,66 @@ def test__resolved_interpreter_windows_pyexe_fails(sysfind, make_one):
     sysfind.assert_any_call("py")
 
 
+@pytest.mark.parametrize(
+    ["input_", "path_result", "expected"],
+    [
+        ("3.7", r"c:\python37-x64\python.exe", r"c:\python37-x64\python.exe"),
+        ("3.7", r"c:\fails\python.exe", None),
+        ("python3.7", r"c:\python37-x64\python.exe", None),
+        ("2.7", r"c:\python37-x64\python.exe", None),
+        ("goofy", r"c:\python37-x64\python.exe", None),
+        ("3.7", None, None),
+        ("python3.7", None, None),
+        ("2.7", None, None),
+        ("goofy", None, None),
+    ],
+)
+@mock.patch("nox.virtualenv._SYSTEM", new="Windows")
+@mock.patch.object(py.path.local, "sysfind")
+def test__resolved_interpreter_windows_path_and_version(
+    sysfind, make_one, input_, path_result, expected
+):
+    # Establish that if we get a standard pythonX.Y path, we look it
+    # up via the path on Windows.
+    venv, _ = make_one(interpreter=input_)
+
+    # Trick the system into thinking that it cannot find python3.6
+    # (it likely will on Unix). Also, we don't give it a dummy
+    # py launcher. But we give it a dummy python interpreter to find
+    # in the system path.
+    if path_result:
+
+        def mock_sysexec(_):
+            if path_result == r"c:\fails\python.exe":
```

Comment:
If this "fails" value is going to be used as a sentinel value like this, could you make it a module global?

Reply:
This is a bit of a hack to signify that sysexec should throw an exception in the parametrization table. I can put it in a constant, but it would essentially still be the same hack, I am open to suggestion on how to handle this kind of more complex parametrization matrix. For now, I hope I can remove it when splitting up the test case as per your review comment above.

Reply:
Yup, no issue with the hack, just wanted to avoid copy-pasting the value in multiple places.